### PR TITLE
Level 4

### PR DESCRIPTION
**Pull request by MaximKalinin**

### CODE REVIEW

File: src/App.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    socket.pipe(socket);
+                    socket.end();
+                    socket.destroy();
+
+                    this._web.send('transfer-completed');
+                });
+            }).catch(() =>
+            {
+                dialog.showMessageBox(
+                {
```
Comment (Pigotz) [inline src/App.ts:76]: Not necessary new line here :see_no_evil:

File: src/App.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            this._web.send('set-port', port);
+        });
+    }
+
+    listen() 
+    {
+        this._server = net.createServer((socket) =>
+        {
+            const file_chunks: Buffer[] = [];
+            socket.on('data', (data) =>
```
Comment (Pigotz) [inline src/App.ts:97]: New line here :see_no_evil:

File: src/App.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        });
+
+        this._web.on('did-finish-load', () =>
+        {
+            const port = this._server.address().port;
+            this._web.send('set-port', port);
+        });
+    }
+
+    listen()
```
Comment (Pigotz) [inline src/App.ts:92]: Missing access modifier and return type

File: src/App.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+        this._server.listen();
+    }
+
+    private _check_connection(client: IClient): Promise<net.Socket>
+    {
+        return new Promise((resolve, reject) =>
+        {
+            let socket: net.Socket;
+            const timeout = setTimeout(() =>
```
Comment (Pigotz) [inline src/App.ts:116]: New line :see_no_evil:

**Review by Pigotz — CHANGES_REQUESTED** [review]

File: src/App.ts (hunk 3 above)
Comment (MaximKalinin) [reply src/App.ts:92]: `public listen(): void`?

**Review by Pigotz — APPROVED** [review]
:+1:

**Pull request closed without merging**